packages/amazonq/test/e2e/inline/inline.test.ts

```diff
@@ -0,0 +1,179 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import assert from 'assert'
+import {
+    assertTelemetry,
+    closeAllEditors,
+    getTestWindow,
+    registerAuthHook,
+    resetCodeWhispererGlobalVariables,
+    TestFolder,
+    toTextEditor,
+    using,
+} from 'aws-core-vscode/test'
+import { RecommendationHandler, RecommendationService } from 'aws-core-vscode/codewhisperer'
+import { Commands, globals, sleep, waitUntil } from 'aws-core-vscode/shared'
+import { loginToIdC } from '../amazonq/utils/setup'
+
+describe('Amazon Q Inline', async function () {
+    let tempFolder: string
+    const waitOptions = {
+        interval: 500,
+        timeout: 10000,
+        retryOnFail: false,
+    }
+
+    before(async function () {
+        await using(registerAuthHook('amazonq-test-account'), async () => {
+            await loginToIdC()
+        })
+    })
+
+    beforeEach(async function () {
+        registerAuthHook('amazonq-test-account')
+        const folder = await TestFolder.create()
+        tempFolder = folder.path
+        await closeAllEditors()
+        await resetCodeWhispererGlobalVariables(false)
+    })
+
+    afterEach(async function () {
+        await closeAllEditors()
+    })
+
+    async function setupEditor({ name, contents }: { name?: string; contents?: string } = {}) {
+        const fileName = name ?? 'test.ts'
+        const textContents =
+            contents ??
+            `function fib() {
+
+
+}`
+        await toTextEditor(textContents, fileName, tempFolder, {
+            selection: new vscode.Range(new vscode.Position(1, 4), new vscode.Position(1, 4)),
+        })
+    }
+
+    async function waitForRecommendations() {
+        const ok = await waitUntil(async () => RecommendationHandler.instance.isSuggestionVisible(), waitOptions)
+        if (!ok) {
+            assert.fail('Suggestions failed to become visible')
+        }
+    }
+
+    async function waitForTelemetry() {
+        const ok = await waitUntil(
+            async () =>
+                globals.telemetry.logger.query({
```

Comment:
Doesn't `assertTelemetry()` do this, or what is it missing?

Reply:
assertTelemetry does the querying but it doesn't wait for the event to appear as far as I can tell. I want the tests to explicitly wait until those telemetry events appear and then finally assert on them